### DISCUSSION TITLE
Restore Previous Immich Library State

### DISCUSSION THREAD

**Discussion by BowChika** [immich-app/immich#3223] [Q&A]
Hi. I self-host immich. I have two accounts on my server; one for me and another for my wife. Between us, we have about 300 GBs of photos uploaded over a few months.

Yesterday, I accidentally brought down the immich container with -v flag  (docker compose down -v) which deleted the (db) volume and hence all the  user data and library metadata. Now, the actual photos and metadata (encoded-video, library, thumbs, upload folders) are still in the original immich folder structure on my server. But when I accessed immich, it  started as a fresh install asking me to setup  admin user and so on. I created (more like replicated) admin user and my wife's user.

Is there a way to re-associate the previous library, meta data, album information etc. to the newly created users? Or just a way to restore to the previous state somehow?

And I learnt my lesson and have included the dbdump container to avoid this issue in the future. But just to trying figure out how to recover from the current mistake.

Thanks!

**Comment by bo0tzz** [comment]
There is no smooth recovery here, as this should be covered by backups ;). You'll need to reimport everything using the CLI.

**Reply by BowChika** [reply]
Thanks. Is there some material on the db structure and what goes where that I can refer to for re-import? Or I need to go through the source?

**Reply by nedlane** [reply]
Hello! I ran into a similar issue, could anybody help me out on what commands I would need to run for the CLI to begin to restore my photos? Any help appreciated!
